hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/MCGetMapConfigCodec.java

```diff
@@ -148,9 +148,20 @@ public static ClientMessage encodeRequest(java.lang.String mapName) {
          * Classname of the SplitBrainMergePolicy for the map.
          */
         public java.lang.String mergePolicy;
+
+        /**
+         * Global indexs of the map.
+         */
+        public java.util.List<com.hazelcast.config.IndexConfig> globalIndexes;
+
+        /**
+         * True if the globalIndexes is received from the member, false otherwise.
+         * If this is false, globalIndexes has the default value for its type.
+         */
+        public boolean isGlobalIndexesExists;
```

Comment:
I can't see any sign of this in the client protocol. Is this hand-written code? It will be overwritten/removed when the codecs are re-generated.

I suggest dropping this boolean, making `globalIndexes` into a nullable list, and handling `null` as a special value when the member doesn't send the indexes.

If you decide keeping this flag instead, then please declare it in the client-protocol, and don't modify generated code. Also I'd pick a better name like `globalIndexesExist` because this name combines singular and plural notation. But overall I suggest picking the ^first suggestion wil the nullable list.

Reply:
This class is autogenerated, I haven't changed it manually.

Reply:
Yet I can't see the same flag on the protocol PR. Maybe you forgot to commit & push it?

Reply:
`globalIndexesExist ` is generated automatically for compatibility based on `globalIndexesExist` list added in the later version.
If `globalIndexes` existed since `2.0` there would be no such flag.
You can checkout the code from `client-protocol` PR and generate the client.

Reply:
I see. is this also generated when `globalIndexes` is marked as nullable in the protocol definition?

Reply:
yes, it is

Reply:
For the record, any parameter added later after the message is defined has an associated `isXXXExists` parameter, to distinguish the old clients/members that do not write that field, since they are not aware of it